web/pages/admin/caregivers.tsx

```diff
@@ -1,42 +1,55 @@
-import React from "react";
-import { deleteCaretaker, getCaregiverPage } from "db/actions/admin/Caregiver";
+import React, { useState } from "react";
+import { addNewCaregiver, deleteCaretaker, getCaregiverPage } from "db/actions/admin/Caregiver";
 import PaginatedTable from "@components/tables/PaginatedTable";
 import { usePaginatedData } from "@components/molecules/Pagination/PaginationHooks";
 import { CAREGIVERS_TAB } from "@lib/utils/consts";
+import ButtonWithIcon from "@components/buttonWithIcon";
+import { FaPlus } from "react-icons/fa";
+import Modal from "@components/modal";
+import CaretakerModal from "@components/modals/CaretakerModal";
 
 const tab = CAREGIVERS_TAB;
 
 export default function genCaregiversTab() {
-    const { data: caretakers, totalRecords, currPage, setCurrPage } = usePaginatedData(getCaregiverPage, tab);
+    const { data: caretakers, totalRecords, currPage, setCurrPage, refresh } = usePaginatedData(getCaregiverPage, tab);
 
     const columns = React.useMemo(() => [
         { Header: "Name", accessor: "name" },
         { Header: "Email", accessor: "email" },
         { Header: "Phone", accessor: "phone" },
+        { Header: "Assigned to Child?", accessor: "assigned"}
     ], []);
 
     const handleDelete = async (caregiver: any) => {
-        deleteCaretaker(caregiver.id).then(() => {
-          alert("Caretaker deleted");
-          setCurrPage(1)
+        deleteCaretaker(caregiver).then(() => {
+          refresh();
+          alert("Caretaker deleted");    
         })
     };
-    
+  
+    const [addModal, toggleAddModal] = useState(false);
     const paginatedProps = {totalRecords: totalRecords, pageNumber: currPage}
     const tableProps = {columns: columns, data: caretakers, onDelete: handleDelete}
 
     return (
-      <div className="max-h-screen overflow-auto w-full">
-        <div className="relative mt-20 border-t w-full " />
-          <div className="pt-6 px-8 flex h-full flex-col justify-left">
-            <div className="flex flex-row justify-between">
-              <div className="flex flex-row">
-                <h1 className="text-2xl mb-5 font-bold">Caregivers</h1>
-                <h2 className="pl-4 pt-2 pb-8 text-sm text-slate-500">
-                  {totalRecords + " People"}
-                </h2>
-              </div>
-            </div>
+      <div>
+      <div className="absolute mt-20 border-t" />
+      <div className="pt-6 px-8 flex h-full flex-col justify-left">
+        <div className="flex flex-row justify-between">
+          <div className="flex flex-row">
+            <h1 className="text-2xl mb-5 font-bold">Caregivers</h1>
+            <h2 className="pl-4 pt-2 pb-8 text-sm text-slate-500">
+              {totalRecords + " People"}
+            </h2>
+          </div>
+          <div>
+            <ButtonWithIcon
+              icon={<FaPlus />}
+              text="Add a caretaker"
```

Comment:
This is an issue even in the Figma but the wording should be "Caregiver" not "Caretaker" 😅 I can fix these instances in the code and make sure that the wording is right in the Figma for the future!